airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -769,3 +793,7 @@ def _change_state(self, key, state, pod_id):
     def end(self):
         self.log.info('Shutting down Kubernetes executor')
         self.task_queue.join()
```

Comment:
I think we had problem here before. task_queue.join() was used but task_done() never called., so I think we could have some hangs when trying to stop gracefully KubernetesScheduler.